docs/control_structures.rst

```diff
@@ -317,6 +318,24 @@ For example,
         def isEven(i):
             return i % Int(2) == Int(0)
 
+PyTeal applies these parameter type annotation constraints when compiling subroutine definitions:
+
+* :any:`ScratchVar` parameters *require* a type annotation.
```

Comment:
Heads up: this will probably change when the ABI-types work is completed. But maybe it's okay to start writing the update now?

Reply:
@tzaffi Thanks for the heads up.  Thoughts are below.  If my opinions don't make sense, let's discuss live.  I'm happy to provide supporting rationale.

* Does my working working assumption match the group's opinion?  
  * Make a best-effort attempt to keep source code and user guide docs in lockstep.  
* Specific to ABI types:  In some cases, I can imagine favoring batching of user guide updates.  However, here I feel we ought keep the docs current.